data-node/src/main/java/org/graylog/datanode/configuration/DatanodeKeystore.java

```diff
@@ -87,31 +95,23 @@ public synchronized boolean exists() {
     }
 
     public synchronized boolean hasSignedCertificate() throws DatanodeKeystoreException {
-        return isSignedCertificateChain(loadKeystore());
+        return !isSelfSignedDatanodeCert(loadKeystore());
     }
 
     public synchronized static boolean isSignedCertificateChain(KeyStore keystore) throws DatanodeKeystoreException {
-        try {
-
-            final Certificate[] certificateChain = keystore.getCertificateChain(DATANODE_KEY_ALIAS);
+        return !isSelfSignedDatanodeCert(keystore);
+    }
 
-            if (certificateChain.length < 2) {
-                // only one cert, it's a self-signed cert!
-                return false;
-            }
-            try {
-                // let's take first cert, which is the datanode. It should be signed by a private key that belongs
-                // to the public key in the second certificate
-                certificateChain[0].verify(certificateChain[1].getPublicKey());
```

Comment:
this check for non self-signed certs is not effectively removed. do we not need it any more?

Reply:
I think this check was an attempt to do the same, in a clumsy and fragile way. For our use case, knowing that the cert is self-signed is enough. Together with checks we added recently, this should provide better overall experience.

Reply:
yes, I think you are right. since we handle the signing, it should be enough to check the negative case that it is self-signed